### PR TITLE
Linking embeddable assemblies, making the interop packages package reference compatible 

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug
Link: Fixes https://github.com/NuGet/Home/issues/5414, Example for https://github.com/NuGet/Home/issues/2365
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: Update our interop package according to the guidance we want to give out to the users, to reference interops via targets. 

## Testing/Validation
Tests Added: No  
Reason for not adding tests:  
Validation done:

### CODE REVIEW

File: src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj [hunk 1]
```diff
@@ -3,23 +3,28 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
 
   <PropertyGroup>
+    <PackageId>NuGet.SolutionRestoreManager.Interop</PackageId>
+    <PackageVersion>$(Version)</PackageVersion>
```
Comment (rohit21agrawal) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj]: this shouldn't be required

File: src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj [hunk 2]
```diff
@@ -3,23 +3,28 @@
...
     <TargetFramework>net46</TargetFramework>
     <Shipping>true</Shipping>
     <PackProject>true</PackProject>
     <IncludeInVsix>true</IncludeInVsix>
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
```
Comment (rohit21agrawal) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj]: this is not supported, remove

File: src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj [hunk 3]
```diff
@@ -10,11 +10,21 @@
...
   <PropertyGroup>
+    <PackageId>NuGet.VisualStudio</PackageId>
+    <PackageVersion>$(Version)</PackageVersion>
     <AssemblyDescription>APIs for invoking NuGet services in Visual Studio.</AssemblyDescription>
+    <Description>APIs for invoking NuGet services in Visual Studio.</Description>
     <ImportedFromTypeLib>NuGet.VisualStudio</ImportedFromTypeLib>
     <Guid>228F7591-2777-47D7-B81D-FEADFC71CEB5</Guid>
     <ComVisible>false</ComVisible>
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
```
Comment (rohit21agrawal) [inline src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj]: remove

File: src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj [hunk 4]
```diff
@@ -10,11 +10,21 @@
   </PropertyGroup>
 
   <PropertyGroup>
+    <PackageId>NuGet.VisualStudio</PackageId>
+    <PackageVersion>$(Version)</PackageVersion>
```
Comment (rohit21agrawal) [inline src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj]: remove

File: src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.targets [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!-- See https://github.com/NuGet/Home/issues/2365 for more info.
+       The decision whether an assembly needs to be linked or referenced belongs to the package author.
+       -->
+  <Target Name="LinkEmbeddableAssemblies" AfterTargets="ResolveReferences" BeforeTargets="FindReferenceAssembliesForReferences">
+    <ItemGroup>
+      <ReferencePath Condition=" '%(FileName)' == 'NuGet.SolutionRestoreManagerInterop' ">
```
Comment (rohit21agrawal) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.targets]: NuGet.SolutionRestoreManager.Interop ?

File: src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj [hunk 6]
```diff
@@ -25,25 +35,11 @@
...
-    <NuspecProperties>version=$(Version);configuration=$(Configuration)</NuspecProperties>
-    <NuspecBasePath>$(OutputPath)</NuspecBasePath>
-  </PropertyGroup>
-
-  <!--
-  <ItemGroup>
-    <None Include="install.ps1">
-      <PackagePath>tools\</PackagePath>
+    <Content Include="NuGet.VisualStudio.targets">
+      <PackagePath>build</PackagePath>
```
Comment (rohit21agrawal) [inline src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj:37]: i am wondering if this needs to be build/TFM and buildCrossTargeting instead so it works for both inner and outer builds
@emgarten suggestions?

**Review by rohit21agrawal — CHANGES_REQUESTED** [review]
some feedback

File: src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj [hunk 7]
```diff
@@ -3,23 +3,28 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
 
   <PropertyGroup>
+    <PackageId>NuGet.SolutionRestoreManager.Interop</PackageId>
+    <PackageVersion>$(Version)</PackageVersion>
     <TargetFramework>net46</TargetFramework>
     <Shipping>true</Shipping>
     <PackProject>true</PackProject>
     <IncludeInVsix>true</IncludeInVsix>
+    <authors>Microsoft</authors>
```
Comment (rohit21agrawal) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj:11]: if at other places the author is NuGet , we should change this to NuGet as well

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj:11]: It's Microsoft for the interops only. 
That makes me hesistant to change it.

File: src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj (hunk 2 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj]: In the package generated nuspec I see the owners metadata. 

Is this a bug or something funny going on?

Comment (rohit21agrawal) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj]: owners defaults to whatever the value of authors is but currently there is no way to set it from a sdk project.

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/NuGet.SolutionRestoreManager.Interop.csproj]: Ah, makes sense. Will remove.

File: src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj (hunk 6 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj:37]: Why would we need it in the outer build?

**Comment by nkolev92** [comment]
Addressed feedback so far. @rohit21agrawal 
Owners & build vs buildCrossTargeting need clarifications.

File: src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj (hunk 6 above)
Comment (rohit21agrawal) [reply src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj:37]: not needed, already discussed with justin

**Review by rohit21agrawal — APPROVED** [review]

**Review by jainaashish — APPROVED** [review]
Looks good!

**Pull request merged**